### PR TITLE
Reintroduce server-side tests in config

### PR DESCRIPTION
**Pull request by mxdvl**
## What does this change?

Reintroduce server side tests location on the config object. Adds a narrower type, which ensures correct usage.

## Why?

We need parity between `frontend` and DCR. It was removed in df58cd0, without fanfare.

### Before

`window.guardian.config.tests` is `undefined`

### After

`window.guardian.config.tests` is `{}`

### CODE REVIEW

File: dotcom-rendering/index.d.ts [hunk 1]
```diff
@@ -688,6 +688,11 @@ interface CommercialConfigType {
 	ampIframeUrl: string;
 }
 
+type ServerSideTests = {
+    [k: `${string}Variant`]: "variant";
+    [k: `${string}Control`]: "control";
```
Comment (oliverlloyd) [inline dotcom-rendering/index.d.ts:693]: Where does `string` come from here?

File: dotcom-rendering/src/model/window-guardian.ts [hunk 2]
```diff
@@ -23,7 +23,7 @@ export interface WindowGuardianConfig {
 		googletag: string;
 	};
 	switches: { [key: string]: boolean };
-	tests?: { [key: string]: string };
+	tests: ServerSideTests;
```
Comment (oliverlloyd) [inline dotcom-rendering/src/model/window-guardian.ts:26]: This was optional before, are we now sure it will always be provided?

Comment (mxdvl) [reply dotcom-rendering/src/model/window-guardian.ts:26]: Yes, from the look of https://www.theguardian.com/uk-news/2021/oct/26/woman-cleared-obstructing-met-police-partner-shot-stun-gun.json?dcr (assuming you’re not in any test)

abTest empty object

Comment (oliverlloyd) [reply dotcom-rendering/src/model/window-guardian.ts:26]: Nice 👍

File: dotcom-rendering/index.d.ts (hunk 1 above)
Comment (mxdvl) [reply dotcom-rendering/index.d.ts:693]: `${string}Variant` is “any string ending in `Variant`”. As a RegEx is would be `/.*Variant/`.

https://github.com/guardian/frontend/blob/0657f594e2fd51125d28df9048c48d6eede26bb6/common/app/experiments/ExperimentsDefinition.scala#L25

And then it’s camelCased a few lines below.

Comment (oliverlloyd) [reply dotcom-rendering/index.d.ts:693]: Gotcha. Thanks for the detailed explanation!

**Review by oliverlloyd — APPROVED** [review]
🚀

**Pull request merged**